### PR TITLE
Update folder structure based on style guide

### LINKED ISSUE #327: Update folder structure based on latest Style Guide
Our current folder structure includes a lot of module paths that are prefixed with `+`.  
See: https://github.com/DSpace/dspace-angular/tree/master/src/app

This naming pattern used to be the recommended one for lazy loaded folders, see: https://github.com/angular/angular.io/blob/b21c90c358129efdd55da897fb7756d44f3b65ae/public/docs/ts/latest/guide/style-guide.jade#L1252

However, in the latest version of the style guide, that recommendation no longer exists: https://angular.io/guide/styleguide#lazy-loaded-folders

We should consider simply renaming these folders to no longer include the `+` prefix, as that's now a non-standard folder name.

### PR DESCRIPTION
**Pull request by artlowel**
## References
* Fixes #327
* Replaces #527

## Description
This PR removes the `+` from all folder names. It was used to indicate lazily loaded modules, but since that's the default now it is no longer recommended in the style guide.

## Instructions for Reviewers
The biggest thing to verify is that I haven't accidentally removed a + where I shouldn't have, e.g. in a string that doesn't refer to a path.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/browse-by/browse-by-switcher/themed-browse-by-switcher.component.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class ThemedBrowseBySwitcherComponent extends ThemedComponent<BrowseBySwi
   }
 
   protected importThemedComponent(themeName: string): Promise<any> {
-    return import(`../../../themes/${themeName}/app/+browse-by/+browse-by-switcher/browse-by-switcher.component`);
+    return import(`../../../themes/${themeName}/app/browse-by/+browse-by-switcher/browse-by-switcher.component`);
```
Comment (tdonohue) [inline src/app/browse-by/browse-by-switcher/themed-browse-by-switcher.component.ts:20]: @artlowel : Still seeing a "+browse-by-switcher" in this line.  That has been renamed but this import needs a quick fix.

Comment (artlowel) [reply src/app/browse-by/browse-by-switcher/themed-browse-by-switcher.component.ts]: Good catch! 
Fixed

File: src/app/logout-page/logout-page.component.scss [hunk 2]
```diff
@@ -0,0 +1 @@
+@import 'src/app/login-page/login-page.component.scss';
```
Comment (tdonohue) [inline src/app/logout-page/logout-page.component.scss:1]: This path changed.  On main it's `@import '../+login-page/login-page.component.scss';`   But this now starts with `src/app`

File: src/app/my-dspace-page/my-dspace-page.component.scss [hunk 3]
```diff
@@ -0,0 +1 @@
+@import 'src/app/search-page/search.component.scss';
```
Comment (tdonohue) [inline src/app/my-dspace-page/my-dspace-page.component.scss:1]: This path also changed.  Used to be `@import '../+search-page/search.component.scss';`

File: src/themes/custom/app/logout-page/logout-page.component.scss [hunk 4]
```diff
@@ -0,0 +1 @@
+@import 'src/themes/custom/app/login-page/login-page.component.scss';
```
Comment (tdonohue) [inline src/themes/custom/app/logout-page/logout-page.component.scss:1]: @artlowel : Another path change

**Review by tdonohue — APPROVED** [review]
👍 Looks great now, @artlowel !  Thanks again.  I've run this locally and done a quick test of most pages of the site (login/logout, clicked around in admin tools, edited every type of object, completed submissin, etc).  As we'd expect, everything appears to be working.  As soon as GitHub CI finishes it's tests/approval, I'll get this merged.

**Pull request merged**